src/pathfora.js

```diff
@@ -505,14 +505,49 @@
       return true;
     },
 
-    pageVisitsChecker: function (pageVisitsRequired, widget) {
-      return (core.pageViews >= pageVisitsRequired);
+    /**
+     * @description Parse url queries as an object
+     * @param {string} url
+     */
+    parseQueries: function (url) {
+      var queries = {};
+      var pieces = url.split('?');
+      if (pieces.length > 1) {
+        pieces = pieces[1].split('&');
+
+        for (var i = 0; i < pieces.length; i++) {
+          var pair = pieces[i].split('=');
+
+          if (pair.length > 1) {
+            queries[pair[0]] = pair[1];
+          }
+        }
+      }
+      return queries;
     },
 
+    /**
+     * @description Compare query params between the url
+     *              the user is visiting and the match
+     *              rule provided
+     * @param {obj} queries
+     * @param {obj} matchQueries
+     */
+    compareQueries: function (queries, matchQueries) {
+      // any query in the matchQueries should exist in the url in the browser
+      for (var key in matchQueries) {
+        if (matchQueries[key] !== queries[key]) {
+          return false;
+        }
+      }
+
```

Comment:
we should also compare key counts here. this has potential to return a false truth if my `queries` has more keys than `matchQueries` but it contains all those present in the `matchQueries`